src/context.rs

```diff
@@ -1,23 +1,21 @@
-/// # tokio-context
-///
-/// Provides two different methods for cancelling futures with a provided handle for cancelling all
-/// related futures, with a fallback timeout mechanism. This is accomplished either with the
-/// `Context` API, or with the `TaskController` API depending on a users needs.
-///
-/// ## Context
-///
-/// Provides Golang like `Context` functionality. A Context in this respect is an object that is passed
-/// around, primarily to async functions, that is used to determine if long running asynchronous
-/// tasks should continue to run, or terminate.
+use std::{future::Future, pin::Pin, sync::Arc, time::Duration};
+use tokio::sync::broadcast::{Receiver, Sender};
+use tokio::sync::{broadcast, Mutex};
+use tokio::time::Instant;
+
+/// An object that is passed around to asynchronous functions that may be used to check if the
+/// function it was passed into should perform a graceful termination.
 ///
-/// You build a new Context by calling its `new()` constructor, which optionally takes a duration
-/// that the context should run for. Calling the `new()` constructor returns the new `Context`
-/// along with a `Handle`. The `Handle` can either have its `cancel()` method called,
-/// or it can simply be dropped to cancel the context.
+/// You build a new Context by calling its [`new`](fn@Context::new)
+/// constructor, which returns the new [`Context`] along with a [`Handle`]. The [`Handle`] can
+/// either have its `cancel` method called, or it can simply be dropped to cancel the context.
```

Comment:
```suggestion
/// either have its [`cancel`](fn@Context::cancel) method called, or it can simply be dropped to cancel the context.
```

Not entirely sure I got the link right

Reply:
Done.